src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const dateValues = date.split(separator);
+  const dateParts = {};
+
+  for (let i = 0; i < dateValues.length; i++) {
+    dateParts[fromFormat[i]] = dateValues[i];
+  }
+
+  if (dateParts['YYYY']) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  } else if (dateParts['YY'] < 30) {
+    dateParts['YYYY'] = `20${dateParts['YY']}`;
+  } else if (dateParts['YY'] >= 30) {
+    dateParts['YYYY'] = `19${dateParts['YY']}`;
```

Comment:
The logic for determining the century based on `YY` is hardcoded to assume that years less than 30 belong to the 2000s and years 30 or greater belong to the 1900s. This might not be flexible enough for all use cases. Consider making this logic more adaptable or configurable.